src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -246,4 +248,36 @@ public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([F
         var responses = secrets.Select(s => new BaseSecretResponseModel(s));
         return new ListResponseModel<BaseSecretResponseModel>(responses);
     }
+
+    [HttpPost("secrets/{organizationId}/move")]
+    public async Task BulkMoveToProjectAsync(Guid organizationId, [FromBody] BulkMoveToProjectsRequestModel request)
+    {
+        var secrets = (await _secretRepository.GetManyByIds(request.Secrets)).ToList();
+        if (!secrets.Any() || secrets.Count != request.Secrets.Count)
+        {
+            throw new NotFoundException();
+        }
+
+        if (secrets.Any(secret => secret.OrganizationId != organizationId) ||
+            !_currentContext.AccessSecretsManager(organizationId))
+        {
+            throw new NotFoundException();
+        }
+
+        foreach (var secret in secrets)
+        {
+            var authorizationResult = await _authorizationService.AuthorizeAsync(User, secret, SecretOperations.Update);
+            if (!authorizationResult.Succeeded)
+            {
+                throw new NotFoundException();
+            }
+        }
+
+        if (request.Projects.Count != 1)
+        {
+            throw new BadRequestException("You are only allowed to assign a single project.");
+        }
```

Comment:
This code was ripped of from https://github.com/bitwarden/server/blob/75cae907e82fa2b2afc0ce7517a18d018ec08daa/src/Api/SecretsManager/Controllers/SecretsManagerPortingController.cs#L70-L73

From an outsiders perspective it was built this way because there is a chance a secret might be in multiple projects down the road. If that is not the case I'd be totally fine with making it singular but if that is indeed the plan then I'd probably want to keep this shape and then remove the exception when that becomes possible.

Reply:
For context, at one point we were implementing having on secret in multiple projects. That is why you see things in the code base as it is now.

A decision was made to only support one project for now. 

In the future, it is possible multiple projects will be implemented but for now it isn't on our roadmap. 

For things like this, I typically follow the YAGNI mantra to avoid the very confusion we're currently experiencing.